src/events/onDelete.ts

```diff
@@ -0,0 +1,44 @@
+import { Message, PartialMessage } from 'discord.js';
+import { ExtendedClient } from '../interfaces/ExtendedClient';
+import { errorHandler } from '../utils/errorHandler';
+import Warnings from '../database/models/Warnings';
+
+export const onDelete = async (
+  bot: ExtendedClient,
+  message: Message | PartialMessage,
+) => {
+  if (message.partial) {
+    try {
+      message = await message.fetch();
+    } catch (error) {
+      return await errorHandler(bot, error, 'fetching partial message');
+    }
+  }
```

Comment:
It's because we can't fetch a partial here.

If the messageDelete event is emitted, the message has already been deleted and we can't fetch it.

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not reusable

Reply:
Thank you 👍 Makes sense